forum/profiles/views.py

```diff
@@ -89,6 +91,67 @@ def list(self, request, *args, **kwargs):
         Handles GET requests for listing startup profiles.
         """
         return super().list(request, *args, **kwargs)
+    
+    @action(detail=False, 
+            methods=['get'], 
+            url_path='viewed', 
+            url_name='viewed-startups',
+            permission_classes=[IsInvestor])
+    def viewed_startups(self, request):
+        """Retrieve a list of recently viewed startups for the authenticated investor"""
+        investor = get_object_or_404(InvestorProfile, user=request.user)
+        viewed_startups = ViewedStratups.objects.filter(investor=investor).order_by('-viewed_at')
```

Comment:
The method fetches all viewed startups for an investor. Add selective prefetching or related field optimization to reduce query overhead.
```
viewed_startups = ViewedStratups.objects.filter(investor=investor).select_related('startup').order_by('-viewed_at')